src/server/server_family.cc

```diff
@@ -270,40 +270,113 @@ error_code ServerFamily::LoadRdb(const std::string& rdb_file) {
   return ec;
 }
 
-void AppendMetric(http::StringResponse* resp, absl::AlphaNum name, absl::AlphaNum val) {
-  /**
-   * Gets a metric name and a value and write it using Prometheus format.
-   *
-   * TODO:
-   *  1. add metrics descriptions.
-   *  2. support other types of metrics, not just gauge :)
-   */
-
-  const auto full_name = StrCat("dragonfly_", name);
-  absl::StrAppend(&resp->body(), "# HELP ", full_name, " ", name, "\n");
-  absl::StrAppend(&resp->body(), "# TYPE ", full_name, " gauge\n");
-  absl::StrAppend(&resp->body(), full_name, " ", val, "\n");
+enum MetricType { COUNTER, GAUGE, SUMMARY, HISTOGRAM };
+
+const char* MetricTypeName(MetricType type) {
+  switch (type) {
+    case MetricType::COUNTER:
+      return "counter";
+    case MetricType::GAUGE:
+      return "gauge";
+    case MetricType::SUMMARY:
+      return "summary";
+    case MetricType::HISTOGRAM:
+      return "histogram";
+  }
+  return "unknown";
+};
+
+#define GET_METRIC_FULLNAME(metric_name) StrCat("dragonfly_", metric_name)
```

Comment:
c macros are usually the last option for me.
it's considered a good style in c++ to avoid them as much as possible. In this case a simple 
inline function would do the same work.